src/daft-minhash/src/minhash.rs

```diff
@@ -0,0 +1,215 @@
+use std::{
+    ops::{Add, BitAnd, Mul, Rem, Shr},
+    simd::{cmp::SimdOrd, Simd},
+};
+
+use common_error::DaftResult;
+use mur3::murmurhash3_x86_32;
+
+// which SIMD to use
+const SIMD_LANES: usize = 8;
+type S = Simd<u64, SIMD_LANES>;
+
+const MERSENNE_PRIME: u64 = (1 << 61) - 1;
+// const MERSENNE_SIMD: S = S::from_array([MERSENNE_PRIME; SIMD_LANES]);
+// const MERSENNE_EXP_SIMD: S = S::from_array([61; SIMD_LANES]);
+const MAX_HASH: u64 = 0xffffffff;
+const MAX_HASH_SIMD: S = S::from_array([MAX_HASH; SIMD_LANES]);
+
+#[inline(always)]
+fn fast_simd_rem(x: S) -> S {
+    (x + x.shr(61)).bitand(MAX_HASH_SIMD)
+}
+
+#[inline(always)]
+fn simd_min(hh: S, aa: &[S], bb: &[S], out: &mut [S]) {
+    let mut h = hh;
+    for ((a, b), o) in aa.iter().zip(bb.iter()).zip(out.iter_mut()) {
+        for _ in 0..SIMD_LANES {
+            *o = fast_simd_rem(h.mul(*a).add(*b)).simd_min(*o);
```

Comment:
Since the output `fast_simd_rem` is always 32 bit, another thing we can do in the future is compute `simd_min` in 32 bit. This should make the operation twice as fast!